modules/account-lib/src/coin/cspr/transaction.ts

```diff
@@ -183,6 +228,36 @@ export class Transaction extends BaseTransaction {
     }
   }
 
+  /**
+   * Set delegate / undelegate fields inside a json representing the tx.
+   *
+   * @param {Record<string, any>} txJson json to modify
+   */
+  setDelegateFieldsInJson(txJson: Record<string, any>): void {
+    if (
+      getDeployType(this.casperTx.session) === TransactionType.StakingLock ||
+      getDeployType(this.casperTx.session) === TransactionType.StakingUnlock
+    ) {
+      const argName = 0;
+      const argValue = 1;
+
+      const delegateValues = new Map();
+      delegateValues.set(
+        TRANSACTION_TYPE,
+        (this.casperTx.session.getArgByName(TRANSACTION_TYPE) as CLValue).asString(),
+      );
+      delegateValues.set('amount', getDelegateAmount(this.casperTx.session));
+      delegateValues.set(DELEGATE_FROM_ADDRESS, getDelegatorAddress(this.casperTx.session));
+      delegateValues.set(DELEGATE_VALIDATOR, getValidatorAddress(this.casperTx.session));
+
+      txJson['deploy']!['session']['ModuleBytes']['args'].forEach(arg => {
```

Comment:
This looks a bit fragile. Is this different from using `txJson.deploy.session.ModuleBytes.args`?

Reply:
It's the same. However, it uses strings.
I'll change this line to use your suggestion in exchange.